content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx

```diff
@@ -152,14 +152,29 @@ The following table shows which referential action each database supports.
 
 | Database    | Cascade | Restrict | NoAction | SetNull | SetDefault |
 | :---------- | :------ | :------- | :------- | :------ | :--------- |
-| PostgreSQL  | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **✔️**     |
+| PostgreSQL  | **✔️**  | **✔️**   | **✔️**   | **✔️**⌘ | **✔️**     |
 | MySQL       | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **✔️**†    |
 | SQLite      | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **✔️**     |
 | SQL Server  | **✔️**  | **❌**‡  | **✔️**   | **✔️**  | **✔️**     |
 | CockroachDB | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **✔️**     |
 | MongoDB††   | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**     |
 
-- † On MySQL, `SetDefault` is ignored and is silently replaced with `NoAction` instead.
+<Admonition type="warning">
+
+Although referential actions are an ANSI SQL feature, some relational databases diverged from the standard, giving rise to a few special cases.
+
+</Admonition>
+
+### Special cases for referential actions
+
+- † On MySQL (using the InnoDB storage engine), `SetDefault` is not supported, with differences depending on the database version. On MySQL 8+, `SetDefault` is considered an alias for `NoAction`.
+  One can define tables using the `SET DEFAULT` referential action, but once that action is triggered at runtime, a foreign key constraint error is raised.
+  On MySQL 5.6+, however, attempting to create one such table definition with the `SET DEFAULT` referential action fails with a syntax error.
+  MariaDB also lacks proper support for `SetDefault`.
```

Comment:
Can we make this more concrete? Not supported on all versions? How?

Reply:
As InnoDB is used by MariaDB by default, and is the only storage we officially support with `provider = "mysql"`, MariaDB also lacks proper support for `SetDefault`. As we do not officially support MariaDB (do we?) and do not know the exact version of MariaDB for which `ON UPDATE/DELETE SET DEFAULT` is a syntax error (like MySQL 5.6+) or a runtime error once the action is triggered (like MySQL 8+), I'd avoid mentioning any version concerning MariaDB

Reply:
I have done a few tests with the reproduction structure documented [here](https://www.notion.so/prismaio/SetDefault-referential-action-on-MySQL-68d7bdbe6fc947cf8829d7ca7dc2b001#635c971f54a04e7fa213ab2a1c43582e).

- `mariadb:10.5` and superior versions behave like `mysql:8`: one can define `SET DEFAULT` referential actions, but it fails once they get triggered
- `mariadb:10.4` and inferior versions behave like `mysql:5.6`: one cannot define `SET DEFAULT` referential actions, as they would trigger syntax errors in `CREATE/ALTER TABLE` statements

Reply:
I have done a few tests with the reproduction structure documented [here](https://www.notion.so/prismaio/SetDefault-referential-action-on-MySQL-68d7bdbe6fc947cf8829d7ca7dc2b001#635c971f54a04e7fa213ab2a1c43582e).

- `mariadb:10.5` and superior versions behave like `mysql:8`: one can define `SET DEFAULT` referential actions, but it fails once they get triggered
- `mariadb:10.4` and inferior versions behave like `mysql:5.6`: one cannot define `SET DEFAULT` referential actions, as they would trigger syntax errors in `CREATE/ALTER TABLE` statements

Reply:
I think we could remind MariaDB users to the descriptions for the two major MySQL versions. Something like:

"""
MariaDB also lacks proper support for `SetDefault`. MariaDB 10.5+ behaves like like MySQL 8+, accepting `SET DEFAULT` declaration but failing at runtime when those actions are triggered. MariaDB 10.4 and previous versions, instead, act like MySQL 5.6+, rejecting `SET DEFAULT` declarations as syntax errors.
"""

Reply:
That is perfect.